FEAT: always use triclinic thermal expansion calculation

ClaireWaters opened the pull request:
Made some changes to the thermal expansion tensor calculation based on a recent meeting 

- Now always compute the full triclinic thermal expansion tensor, report it, but also report a symmetry-reduced version that enforces zero-valued components and equivalent components based on space group.
- Made the individual components of the property optional, reported based on space group
- Each temperature writes out an instance of crystal-structure-npt to results.edn
- Each temperature writes out a poscar file
- fixed several arithmetic errors in the uncertainty propagation

ClaireWaters committed:
FEAT: always use triclinic thermal expansion calculation

simplify for other structures and report symmetry reduced

PhilippHoellmer commented:
Changes mostly look good to me. One question:

- You say that each temperature writes out a poscar file. However, you call `self.poscar = self.atoms.write ...` after the loop?

PhilippHoellmer committed:
Small change

ClaireWaters committed:
BUG: write poscar for each equilibrated structure

fix arithmetic error in alpha23_err calculation
write logs for lammps file read test to different file name
remove unused variables

brendonwaters commented:
I'm not sure that self.poscar actually gets used for anything, so having it be None shouldn't cause any problems. I wouldn't even bother to initialize it as a class variable except that Ilia's example driver does so too. 

As for being outside the loop, that was indeed a bug. For reasons I don't quite understand, it was writing the correct number of poscar files, but they were all identical copies of the same one, so I didn't notice at first. I've pushed a new commit that moved it into the loop, and made sure to have atoms_reduced write out the poscar file, so we're saving the final equilibrated structure instead of the input structure.

Besides that, I fixed another arithmetic error in the uncertainty propagation, and made sure that the lammps file read test writes a log file to a different file name than the real lammps run, to avoid overwriting data.

PhilippHoellmer committed:
Merge branch 'update_crystal_genome' of https://github.com/openkim-hackathons/HeatCapacityPhonon into update_crystal_genome

PhilippHoellmer committed:
Fixes in test driver

PhilippHoellmer committed:
Updated heat capacity property

PhilippHoellmer committed:
Code for all heat capacities

PhilippHoellmer committed:
Add restart file

PhilippHoellmer commented:
I did not change branches before making the following changes that should now finish development of our test driver:

- Use standard temperature_K and cell_cauchy_stress_eV_angstrom3 arguments during initialization instead of our own arguments + some more checks of the initialization arguments.
- The crystal-structure-npt property is written for every file including a restart file.
- Heat capacity is now reported per atom, per formula, and per mass.

Note that in the first commit "Fixes in test driver", I replaced `crystal_genome_designation["prototype_label"]` by `self.prototype_label` because `crystal_genome_designation` is not necessarily defined.

Please check if you agree with this final version of the test driver.

PhilippHoellmer committed:
Changed comment

PhilippHoellmer committed:
Updated gitignore

PhilippHoellmer committed:
Extended properties to include coordinates-file and crystal-genome-source-structure-id

PhilippHoellmer committed:
Updated gitignore

PhilippHoellmer committed:
Made test driver a package

PhilippHoellmer committed:
Reverted accidental change

PhilippHoellmer committed:
Updated main

PhilippHoellmer committed:
Fixed a bug

PhilippHoellmer committed:
Final change

PhilippHoellmer committed:
Updated gitignore

PhilippHoellmer commented:
I just made some final changes that all files are created in the output directory relative to our test_driver.py file. Also, the resulting properties are written to a file in the end. All of this allowed to run our test driver as a package `python -m test_driver.test_driver` without any error.

I'm going to wait for @brendonwaters review of all of these changes and then merge back into main.

PhilippHoellmer committed:
Typo

PhilippHoellmer committed:
Removed comment

PhilippHoellmer committed:
Final bug

thomas-egg committed:
Add driver and job file for running several tests

PhilippHoellmer committed:
All files are now written in the current directory

PhilippHoellmer committed:
Made arguments required

ClaireWaters committed:
ENH: make HeatCapacity TD a kim test driver

add pipeline infrastructure to generate tests from the driver,
and the parameters needed for it.

ClaireWaters committed:
Merge branch 'update_crystal_genome' of github.com:openkim-hackathons/HeatCapacityPhonon into update_crystal_genome

brendonwaters commented:
I'll have to look at your changes more closely before the final merge, but I just added the infrastructure to make this code a real test-driver to this pull request. It now has the template files needed for kimgenie to generate the tests from the driver, and running a few of the tests worked in the KDP. I think we're just about done with this driver, pending a final review and merge of the code.

PhilippHoellmer committed:
Removed output directory from repository

PhilippHoellmer committed:
Made test driver runnable from everywhere

PhilippHoellmer committed:
Removed main part from test driver

PhilippHoellmer committed:
Removed obsolete argument

PhilippHoellmer committed:
Temporary run file

PhilippHoellmer committed:
Undid accidental change

PhilippHoellmer committed:
Removed unused import

PhilippHoellmer committed:
Cleanup

PhilippHoellmer committed:
Path in run_lengh_control

PhilippHoellmer committed:
Fixed writing of accuracies file

PhilippHoellmer committed:
Fixed path

PhilippHoellmer committed:
Kept accuracies file around

PhilippHoellmer committed:
Dump Lammps trajectory

PhilippHoellmer committed:
Write poscar file on error

PhilippHoellmer committed:
Fixed check

PhilippHoellmer committed:
Started general implementation of PCA and KStest

PhilippHoellmer committed:
Updated plots

PhilippHoellmer committed:
Small change

PhilippHoellmer committed:
Add p value output

PhilippHoellmer committed:
Copied over runner and pipeline.stdin.tpl.genie

PhilippHoellmer committed:
Updated test driver

PhilippHoellmer committed:
Revert accidental change

PhilippHoellmer committed:
Bugfix

PhilippHoellmer committed:
Included msd threshold argument

thomas-egg committed:
Change LAMMPS command
